runelite-client/src/main/java/net/runelite/client/ui/components/DragAndDropReorderPane.java

```diff
@@ -0,0 +1,250 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.ui.components;
+
+import java.awt.Component;
+import java.awt.Container;
+import java.awt.LayoutManager;
+import java.awt.Point;
+import java.awt.dnd.DragSource;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.util.HashMap;
+import java.util.Map;
+import javax.swing.BoxLayout;
+import javax.swing.JLayeredPane;
+
+/**
+ * Pane which allows reordering its components via drag and drop.
+ * <p>
+ * For child components' popup menus, try avoid using setComponentPopupMenu as this consumes the MouseEvents required
+ * for drag and drop reordering. Instead, implement the PopupMenuOwner interface in the children components to enable
+ * both child popup menus and drag and drop reordering.
+ */
+public class DragAndDropReorderPane extends JLayeredPane
+{
+	private Point dragStartPoint = null;
+	private Component draggingComponent = null;
+	private int dragIndex = -1;
+
+	private final Map<Integer, PopupMenuOwner> popupMenuCandidates = new HashMap<>();  // keyed by mouse button
+
+	public DragAndDropReorderPane()
+	{
+		super();
+		setLayout(new DragAndDropReorderLayoutManager());
+		MouseAdapter mouseAdapter = new DragAndDropReorderMouseAdapter();
+		addMouseListener(mouseAdapter);
+		addMouseMotionListener(mouseAdapter);
+	}
+
+	@Override
+	public void setLayout(LayoutManager layoutManager)
+	{
+		if (layoutManager != null && !(layoutManager instanceof DragAndDropReorderLayoutManager))
+		{
+			throw new IllegalArgumentException("DragAndDropReorderPane only supports DragAndDropReorderLayoutManager");
+		}
+		super.setLayout(layoutManager);
+	}
+
+	private void startDragging(Point point)
+	{
+		draggingComponent = getDefaultLayerComponentAt(dragStartPoint);
+		if (draggingComponent == null)
+		{
+			dragStartPoint = null;
+			return;
+		}
+		dragIndex = getPosition(draggingComponent);
+		setLayer(draggingComponent, DRAG_LAYER);
+		moveDraggingComponent(point);
+	}
+
+	private void drag(Point point)
+	{
+		moveDraggingComponent(point);
+
+		Component component = getDefaultLayerComponentAt(point);
+		if (component != null)
+		{
+			int index = getPosition(component);
+			dragIndex = index < dragIndex ? index : index + 1;
+			revalidate();
+		}
+	}
+
+	private void finishDragging()
+	{
+		if (draggingComponent != null)
+		{
+			setLayer(draggingComponent, DEFAULT_LAYER, dragIndex);
+			revalidate();
+		}
+		dragStartPoint = null;
+		draggingComponent = null;
+		dragIndex = -1;
+	}
+
+	private void moveDraggingComponent(Point point)
+	{
+		// place the center of the dragging component onto the mouse cursor
+		int y = point.y - draggingComponent.getHeight() / 2;
+		// clamp the height to stay within the pane
+		y = Math.max(y, 0);
+		y = Math.min(y, getHeight() - draggingComponent.getHeight());
+
+		Point location = new Point(0, y);
+		draggingComponent.setLocation(location);
```

Comment:
Remove the `location` variable above and set the location with the new Point directly.
```suggestion
		draggingComponent.setLocation(new Point(0, y));
```

Reply:
Done